webpack/assets/javascripts/react_app/common/DeprecationService.js

```diff
@@ -0,0 +1,22 @@
+/* eslint-disable no-console, max-len */
+
+export const deprecate = (oldMethod, newMethod, version = '1.22') =>
```

Comment:
IMHO this shouldn't have a default since you always need to think about when it'll be deprecated. We deprecate over 2 releases so that means after the next version is released you'd need to update all invocations anyway if you want to update the default.

Reply:
Totally agree with you but not in this PR please, will need to search for places using it to make sure no one using the default value.
I created an issue for that: https://projects.theforeman.org/issues/26386

Or let's deprecate the deprecate function? :/